DigitalLearningSolutions.Data.Tests/Services/PasswordResetServiceTests.cs

```diff
@@ -1,67 +1,218 @@
+            (
+                (
+                    new List<AdminUser> { UserTestHelper.GetDefaultAdminUser() },
+                    new List<DelegateUser> { UserTestHelper.GetDefaultDelegateUser() }
+                ));
+
+            passwordResetService = new PasswordResetService(
+                userService,
+                passwordResetDataService,
+                logger,
+                emailService,
+                clockService);
+        }
+
+        [Test]
+        public void Trying_get_null_user_should_throw_an_exception()
+        {
+            // Given
+            A.CallTo(() => userService.GetUsersByEmailAddress(A<string>._))
+                .Returns((new List<AdminUser>(), new List<DelegateUser>()));
+
+            // Then
+            Assert.Throws<UserAccountNotFoundException>(
+                () => passwordResetService.GenerateAndSendPasswordResetLink("recipient@example.com", "example.com"));
+        }
+
+        [Test]
+        public void Trying_to_send_password_reset_sends_email()
+        {
+            // Given
+            var emailAddress = "recipient@example.com";
+            var adminUser = Builder<AdminUser>.CreateNew().With(user => user.EmailAddress = emailAddress)
+                .Build();
+
+            A.CallTo(() => userService.GetUsersByEmailAddress(A<string>._))
+                .Returns((new[] { adminUser }.ToList(), new List<DelegateUser>()));
+
+            // When
+            passwordResetService.GenerateAndSendPasswordResetLink(emailAddress, "example.com");
+
+            // Then
+            A.CallTo(
+                    () =>
+                        emailService.SendEmail(
+                            A<Email>.That.Matches(
+                                e =>
+                                    e.To[0] == emailAddress &&
+                                    e.Cc.IsNullOrEmpty() &&
+                                    e.Bcc.IsNullOrEmpty() &&
+                                    e.Subject == "Digital Learning Solutions Tracking System Password Reset"))
+                )
+                .MustHaveHappened();
+            ;
+        }
+
+        [Test]
+        public async Task Reset_password_hash_is_invalid_if_125_minutes_old()
+        {
+            // Given
+            var createTime = DateTime.UtcNow;
+
+            var resetPasswordModel = Builder<ResetPassword>.CreateNew()
+                .With(m => m.Id = 1)
+                .With(m => m.PasswordResetDateTime = createTime)
+                .With(m => m.ResetPasswordHash = "Old Hash Brown")
+                .Build();
+            A.CallTo(() => passwordResetDataService.FindAsync(1)).Returns(Task.FromResult(resetPasswordModel));
+
+            var candidate = Builder<DelegateUser>.CreateNew()
+                .With(user => user.ResetPasswordId = resetPasswordModel.Id)
+                .Build();
+            A.CallTo(() => userService.GetUsersByEmailAddress(A<string>._))

```

Comment:
Super minor, but could we specify the email as an argument here (I think it could also benefit the tests above) to make sure the mock only returns if called with the correct email address?